### DISCUSSION TITLE
Add `InternalsVisibleTo` for unit tests

### DISCUSSION THREAD

**Discussion by GeorgeLeithead** [CommunityToolkit/Maui#573] [General]
Currently any `internal` classes, methods or properties can't be tested via Unit tests.  could we look to add to the `AssemblyInfo.shared.cs` for all projects the following, so that we can then implement unit tests for these `internal`s?

```csharp
[assembly: InternalsVisibleTo("CommunityToolkit.Maui.UnitTests")]
```

**Comment by TheCodeTraveler** [comment]
We made a choice not to use `InternalsVisibleTo` to ensure that we made things `public` that were important to test. 

This thought is that if we need it for a test, it should be public. 

Another option is to create a Mock implementation of an interface for the tests: https://github.com/CommunityToolkit/Maui/tree/main/src/CommunityToolkit.Maui.UnitTests/Mocks

**Reply by GeorgeLeithead** [reply]
Thanks for the enlightenment.  I was trying to improve Code Coverage testing.  I'll have a look at the `Mocks`.

**Comment by CliffAgius** [comment]
Closed as answered
